scripts/pylib/twister/twisterlib/environment.py

```diff
@@ -947,6 +947,22 @@ def __init__(self, options=None, default_options=None) -> None:
             if snippet_root:
                 self.snippet_roots.append(Path(module.project) / snippet_root)
 
+
+        self.soc_roots = [Path(ZEPHYR_BASE), Path(ZEPHYR_BASE) / 'subsys' / 'testsuite']
+        self.dts_roots = [Path(ZEPHYR_BASE)]
+        self.arch_roots = [Path(ZEPHYR_BASE)]
+
+        for module in modules:
+            soc_root = module.meta.get("build", {}).get("settings", {}).get("soc_root")
+            if soc_root:
+                self.soc_roots.append(os.path.join(module.project, soc_root))
+            dts_root = module.meta.get("build", {}).get("settings", {}).get("dts_root")
+            if soc_root:
+                self.dts_roots.append(os.path.join(module.project, dts_root))
+            arch_root = module.meta.get("build", {}).get("settings", {}).get("arch_root")
+            if arch_root:
+                self.arch_roots.append(os.path.join(module.project, arch_root))
```

Comment:
**suggestion:** append with **path-like object** ([PEP 519](https://peps.python.org/pep-0519/)):
```suggestion
                self.arch_roots.append(Path(os.path.join(module.project, arch_root)))
```

otherwise following line will cause `AttributeError: 'str' object has no attribute '_closed'`:
https://github.com/zephyrproject-rtos/zephyr/blob/de0f972a4825b5c4ddbe2b3752298b261193f8ea/scripts/pylib/twister/twisterlib/testplan.py#L408

full traceback:
```
Traceback (most recent call last):
  File "./zephyr/scripts/twister", line 215, in <module>
    ret = main(options, default_options)
  File "/.../workspace/zephyr/scripts/pylib/twister/twisterlib/twister_main.py", line 124, in main
    tplan.discover()
  File "/.../workspace/zephyr/scripts/pylib/twister/twisterlib/testplan.py", line 197, in discover
    self.add_configurations()
  File "/.../workspace/zephyr/scripts/pylib/twister/twisterlib/testplan.py", line 410, in add_configurations
    v1_boards = list_boards.find_boards(lb_args)
  File "/.../workspace/zephyr/scripts/list_boards.py", line 117, in find_boards
    return sorted(itertools.chain(*find_arch2board_set(args).values()),
  File "/.../workspace/zephyr/scripts/list_boards.py", line 122, in find_arch2board_set
    arches = sorted(find_arches(args))
  File "/.../workspace/zephyr/scripts/list_boards.py", line 138, in find_arches
    for root in unique_paths(args.arch_roots):
  File "/.../workspace/zephyr/scripts/list_hardware.py", line 182, in unique_paths
    yield from dict.fromkeys(map(Path.resolve, paths)).keys()
  File "/usr/lib/python3.8/pathlib.py", line 1179, in resolve
    if self._closed:
```